vault/router.go

```diff
@@ -62,6 +62,13 @@ func (r *Router) Mount(backend logical.Backend, prefix string, mountEntry *Mount
 	if existing, _, ok := r.root.LongestPrefix(prefix); ok && existing != "" {
 		return fmt.Errorf("cannot mount under existing mount '%s'", existing)
 	}
+	// If this is a secret backend, check to see if the prefix conflicts
+	// with an existing mountpoint
+	if prefix != "" {
+		if match := r.matchingPrefixInternal(prefix); match != "" {
```

Comment:
Let me know how to proceed. I can add some context on the upgrade docs, or is handling the data migration something we want to consider?